### ISSUE TITLE
meson 0.48.0: file-roller fails to generate

### ISSUE THREAD

**Issue by heftig** [mesonbuild/meson#4257]
file-roller uses a `custom_target` as the input for `i18n.merge_file`:

https://gitlab.gnome.org/GNOME/file-roller/blob/7962fbaf86db77ec6c0713491920341487be1681/data/meson.build#L22

Since meson 0.48.0, this fails. Probably caused by 176af2c8c33e79732680ebb1e2a25b5cd7a51bb9.

**Log:**
```
+ exec meson setup --prefix /usr --libdir /usr/lib --libexecdir /usr/lib --bindir /usr/bin --sbindir /usr/bin --includedir /usr/include --datadir /usr/share --mandir /usr/share/man --infodir /usr/share/info --localedir /usr/share/locale --sysconfdir /etc --localstatedir /var --sharedstatedir /var/lib --buildtype release --auto-features enabled --wrap-mode nofallback -D b_lto=true file-roller build
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 364, in run
    app.generate()
  File "/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 136, in generate
    self._generate(env)
  File "/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 165, in _generate
    intr.run()
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3836, in run
    super().run()
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 380, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 404, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 398, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 409, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 705, in function_call
    return func(node, posargs, kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 273, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 162, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3422, in func_subdir
    self.evaluate_codeblock(codeblock)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 404, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 398, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 413, in evaluate_statement
    return self.method_call(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 741, in method_call
    return obj.method_call(method_name, args, kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 1622, in method_call
    value = fn(state, args, kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 219, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 162, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/modules/i18n.py", line 89, in merge_file
    ifile_abs = inputfile.absolute_path(state.environment.source_dir,
AttributeError: 'CustomTargetHolder' object has no attribute 'absolute_path'
The Meson build system
Version: 0.48.0
Source dir: /build/file-roller/src/file-roller
Build dir: /build/file-roller/src/build
Build type: native build
Project name: file-roller
Project version: 3.30.1
Appending CFLAGS from environment: '-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt'
Appending LDFLAGS from environment: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
Appending CPPFLAGS from environment: '-D_FORTIFY_SOURCE=2'
Native C compiler: cc (gcc 8.2.1 "cc (GCC) 8.2.1 20180831")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Program postinstall.py found: YES (/build/file-roller/src/file-roller/postinstall.py)
Library m found: YES
Dependency threads found: YES 
Found pkg-config: /usr/bin/pkg-config (1.5.3)
Dependency glib-2.0 found: YES 2.58.0
Dependency gthread-2.0 found: YES 2.58.0
Dependency gtk+-3.0 found: YES 3.24.1
Dependency libnautilus-extension found: YES 3.30.0
Dependency json-glib-1.0 found: YES 1.4.4
Dependency libnotify found: YES 0.7.7
Dependency libarchive found: YES 3.3.3
Checking for function "mkdtemp" : YES
Program gcpio found: NO
Program cpio found: NO
Configuring config.h using configuration
Compiler for C supports arguments -Wall: YES
Program set-mime-type-entry.py found: YES (/build/file-roller/src/file-roller/data/set-mime-type-entry.py)
```

**Comment by jbicha** [comment]
There was a fix pushed to master for #4304 so I guess this bug should be marked a duplicate of that one.

**Comment by nirbheek** [comment]
Duplicate of https://github.com/mesonbuild/meson/issues/4304
